torch/_dynamo/variables/user_defined.py

```diff
@@ -442,6 +446,17 @@ def call_function(
             else:
                 var_cls = GenericContextWrappingVariable
 
+            if self.value is contextlib._GeneratorContextManager and isinstance(
+                args[0], BaseUserFunctionVariable
+            ):
+                # Replace UserFunctionVariable by GeneratorFunction if the function
```

Comment:
GeneratorFunction -> ContextlibContextManagerFunctionVariable ?

Reply:
Wait, ContextlibContextManagerFunctionVariable is actually supposed to be... FunctionDecoratedByContextlibContextManagerVariable ?

Reply:
> GeneratorFunction -> ContextlibContextManagerFunctionVariable ?

I'll update it.

> Wait, ContextlibContextManagerFunctionVariable is actually supposed to be... FunctionDecoratedByContextlibContextManagerVariable ?

Yes